### PR TITLE
[None][feat] Enable MTP for Step-3.7 NVFP4 and port Step-3.7VL vision tower to TRT-LLM modules

### PR DESCRIPTION
**Pull request by kaiyux**
## Description

Enables Multi-Token Prediction (MTP) speculative decoding for the **Step-3.7-Flash NVFP4** checkpoint and ports the **Step-3.7VL** vision tower and MLP onto TRT-LLM's native modules. Bundled changes:

- **Enable MTP for Step-3.7 NVFP4 + fix spec token ids on the VLM path.** The NVFP4 ModelOpt export ships the 3 plain-path MTP layers (45..47) under the bare `model.layers.*` namespace, so the NVFP4 checkpoint can now run speculative decoding like the FP8/BF16 checkpoints. On the multimodal path the main model consumes fused `inputs_embeds` and `input_ids` is `None`, so the VLM wrapper now forwards the pre-fusion token ids via a new `spec_input_ids` argument (image tokens remapped into vocab range) so the MTP/spec worker always receives valid token ids.

- **Port the Step-3.7VL vision tower to the TRT-LLM `Attention` module.** Replaces the raw-torch SDPA vision attention so the vision encoder runs through the backend FMHA dispatch (context-only, `FULL` mask, varlen via per-image `attn_metadata`). The PerceptionEncoder `head_dim` (96) is not a trtllm-gen FMHA cubin size, so q/k/v/o head dims are zero-padded to 128 while 2D RoPE runs on the real channels; a compensating `q_scaling` preserves the softmax scale. Fused HF `in_proj_*`/`out_proj` weights are remapped onto `qkv_proj`/`o_proj` during weight loading.

- **Port the Step-3.7VL vision MLP to the base TRT-LLM `MLP` module.** Dispatches the non-gated vision FFN through `up_proj -> activation -> down_proj` instead of a bespoke `c_fc`/`c_proj` `nn.Module`. HF `mlp.c_fc`/`mlp.c_proj` weights are remapped onto `up_proj`/`down_proj` during weight loading.

- **Resolve MTP layer count across checkpoint field names.** `update_spec_config_from_model_config` previously read the MTP layer count only from `num_nextn_predict_layers` (DeepSeek-style). Qwen3Next-style configs (including Qwen3.5) expose it as `mtp_num_hidden_layers`; the resolver now reads whichever field is present and falls back to a single shared MTP / EAGLE layer when neither exists.

- **Optimize vision RoPE and multimodal H2D copies.** Precomputes the vision tower's 2D-RoPE `cos`/`sin` tables once per forward pass and threads them through all 47 blocks instead of re-evaluating `cos()`/`sin()` inside every attention call. Restricts multimodal device copies to only the tensors the vision tower actually consumes on GPU. Skips the `spec_input_ids` OOV-sentinel rewrite when speculative decoding is disabled.

## Test Coverage

Unit tests:
- `tests/unittest/_torch/modeling/test_modeling_step3p7.py` — refreshed for the NVFP4 MTP layer layout; removed `test_config_and_weight_accounting_1_nvfp4` waive.
- `tests/unittest/_torch/modeling/test_modeling_step3p7vl.py` — vision tower / MLP TRT-LLM module port and weight remapping, RoPE precomputation path.

Accuracy / integration tests (parametrized over `mtp_nextn` ∈ {0, 3}):
- `accuracy/test_llm_api_pytorch.py::TestStep3_7::test_nvfp4`
- `accuracy/test_llm_api_pytorch.py::TestStep3_7::test_fp8_block_scales`
- `accuracy/test_llm_api_pytorch_multimodal.py::TestStep3_7::test_fp8_block_scales`
- `accuracy/test_llm_api_pytorch_multimodal.py::TestStep3_7::test_nvfp4`

Accuracy reference updates:
- `references/gsm8k.yaml`: added NVFP4 + MTP entry (accuracy 88, reuses non-spec baseline).
- `references/mmmu.yaml`: updated FP8 and NVFP4 baselines from 64 → 60; added FP8+MTP and NVFP4+MTP entries.

Test list updates:
- `tests/integration/test_lists/test-db/l0_dgx_b200.yml`: moved NVFP4 MTP case to QA.
- `tests/integration/test_lists/qa/llm_function_core.txt`: added Step-3.7 / Step-3.7VL NVFP4 + FP8 MTP accuracy cases (both text and multimodal, mtp_nextn=0 and 3).
- `tests/integration/test_lists/waives.txt`: removed `test_config_and_weight_accounting_1_nvfp4` skip (NVFP4 MTP now works).

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of my knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work)).
- If PR introduces API changes, an appropriate PR label is added.
- Any new dependencies have been scanned for license and vulnerabilities.
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes.
- Documentation updated as needed.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`

### CODE REVIEW

**Review by ruodil — APPROVED** [review]

File: tensorrt_llm/_torch/models/modeling_step3p7vl.py
Comment (yechank-nvidia) [inline tensorrt_llm/_torch/models/modeling_step3p7vl.py:1214]: `Step3p7VLForConditionalGeneration` should define `multimodal_data_device_paths`.

Without it, model_engine moves every tensor under multimodal_data to CUDA. For Step3.7VL, only these fields need to be on GPU:

```
  @property
  def multimodal_data_device_paths(self) -> List[str]:
      return [
          "image.pixel_values",
          "image.patch_pixel_values",
          "multimodal_embedding",
      ]
```

`num_patches` and `patch_newline_mask` are metadata, not vision inputs. Keeping them on CPU avoids unnecessary H2D copies and prevents a possible GPU sync if num_patches ever reaches `Step3p7VisionTower.forward()` as a CUDA tensor and hits .tolist().

File: tensorrt_llm/_torch/models/modeling_step3p7vl.py [hunk 1]
```diff
@@ -1003,7 +1325,15 @@ def forward(
             )
             mm_embeds = find_input_mm_embeds(mm_embeds, mm_context_params)
 
+        spec_input_ids = input_ids
         if input_ids is not None and mm_embeds:
+            vocab_size = self.llm.model.embed_tokens.num_embeddings
+            image_token_id = int(getattr(self.config, "image_token_id", 128001))
+            spec_input_ids = torch.where(
```
Comment (yechank-nvidia) [inline tensorrt_llm/_torch/models/modeling_step3p7vl.py]:
```suggestion
            if self.spec_worker is not None and input_ids is not None and mm_embeds:
         spec_input_ids = torch.where(...)
```

Comment (yechank-nvidia) [reply tensorrt_llm/_torch/models/modeling_step3p7vl.py]: Maybe we can skip `torch.where` operations when spec_dec is not ON.

File: tensorrt_llm/_torch/models/modeling_step3p7vl.py
Comment (yechank-nvidia) [inline tensorrt_llm/_torch/models/modeling_step3p7vl.py]: minor pref comment:

`_apply_rotary_emb` recomputes freqs.cos() and freqs.sin() for both q and k in every vision layer. For Step3.7 this is 47 layers, so the same RoPE table is recomputed 94 times per forward. 

The cache footprint is small: for the real 52x52 grid and head_dim=96, storing both cos and sin is about 2 MiB in fp32. 

Consider caching cos/sin per grid, or storing the buffer as cos/sin directly.

**Review by yechank-nvidia — APPROVED** [review]
LGTM. No major blocking components. Thanks for addressing previous PR's comments.

File: tensorrt_llm/_torch/models/modeling_step3p7vl.py
Comment (kaiyux) [reply tensorrt_llm/_torch/models/modeling_step3p7vl.py:1214]: Updated.

File: tensorrt_llm/_torch/models/modeling_step3p7vl.py (hunk 1 above)
Comment (kaiyux) [reply tensorrt_llm/_torch/models/modeling_step3p7vl.py]: Updated.

File: tensorrt_llm/_torch/models/modeling_step3p7vl.py
Comment (kaiyux) [reply tensorrt_llm/_torch/models/modeling_step3p7vl.py]: Updated, the encoder computes `cos` and `sin` once per forward and threads them through every layer now.

**Review by StanleySun639 — APPROVED** [review]

**Comment by xinhe-nv** [comment]
@coderabbitai why didn't you review the code as a QA engineer at the first commit?

**Comment by xinhe-nv** [comment]
@coderabbitai review the pr as a QA engineer to check the test coverage.
